src/Uno.UI/DataBinding/BindingPath.BindingItem.cs

```diff
@@ -540,6 +538,17 @@ public void NewValue()
 				public void NewValue(DependencyObject dependencyObject, DependencyPropertyChangedEventArgs args)
 					=> NewValue();
 			}
+
+			/// <remarks>
+			/// This is defined so that 2 BindingItems are equal if they refer to the same property on the same object
+			/// even if they're a part of 2 different "chains".
+			/// </remarks>
+			public bool Equals(BindingItem? that)
+			{
+				return that != null
+					&& !DependencyObjectStore.AreDifferent(this.DataContext, that.DataContext)
+					&& this.PropertyName.Replace(")", "").Replace("(", "").Split(':', '.')[^1] == that.PropertyName.Replace(")", "").Replace("(", "").Split(':', '.')[^1];
```

Comment:
What's blocking this PR is this line. We need to find a way to compare two `BindingItem.PropertyNam`s and figure out if they point to the same thing. E.g. `(Microsoft.UI.Xaml.Controls:Border.Background)` and `Background` are the same thing. There are also expressions with indexers and some other weird stuff.